build.gradle

```diff
@@ -76,4 +76,132 @@ project(':configurator') {
+            def jreFileName = jreFileNameTemplate(sys,jreFileExtension)
+            def jreFile = file("$jreDownloadDir.absolutePath/$jreFileName")
+            if(!jreFile.exists()){
+                new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/$jreFileName")
+                        .withInputStream {
+                    jreFile.newOutputStream() << it
+                }
+                final hash = MessageDigest.getInstance('SHA-256').with {
+                    jreFile.eachByte(1024) {buff,len ->
+                        update(buff, 0, len)
+                    }
+                    digest().encodeHex() as String
+                }
+                if(hash != jreHashes[index]) {
+                    throw new RuntimeException("Hash mismatch for downloaded $sys JRE. File may have been modified.")
+                }
+            }
+        }
+        
+    }
+}
+
+task unzipJre(description: 'Unzip JREs only when the extract directory is not existing', dependsOn:downloadJre) {
+    doFirst {
+        fileTree(jreDownloadDir).matching {
+            exclude 'jre'
+        }.each { jreFile ->
+                 if(jreFile.name.endsWith('zip')) {
+                     copy {
+                         from zipTree(jreFile)
+                         into jreExtractSubDir('windows')
+                     }
+                 }
+                 else {
+                     copy {
+                         from tarTree(resources.gzip(jreFile))
+                         into jreExtractSubDir(jreFile.name.contains('mac') ? 'mac' : 'linux')
+                     }
+                 }
+        }
+    }
+}
+
+unzipJre.onlyIf {
+    !file(jreExtractDir).exists()
+}
+
+task createPackages(description: 'Create three packages for Linux, Windows and Mac', dependsOn:unzipJre) {
+    //This task also depends on other three tasks which are created by iterating osNames
+}
+
+osNames.each {String sys ->
+    def packageDir = new File("$buildDir/distributions/packages")
+    def (taskType, fileExtension, fileCompression, scriptFile) = setPackageProperties(sys)
+    def taskCLass = Class.forName(taskType)
+    def jarDir = file("$buildDir/libs")
+    def scriptDir = file("$projectDir/launcher-scripts")
+    task "packageFor$sys"(type: taskCLass){
+        mustRunAfter unzipJre
+        archiveBaseName = "admin-console-package-for-$sys"
+        extension = fileExtension
+        if(fileCompression) compression = Compression.GZIP
+        destinationDirectory = packageDir
+        from jarDir
+        from file(jreExtractSubDir(sys))
+        exclude (['*.zip', '*.tar.gz'])
+        from file("${scriptDir.getAbsolutePath()}/$scriptFile")
+
+        doLast {
+            println("Package for $sys is created under ${packageDir.absolutePath}")
+        }
+    }
+    createPackages.dependsOn("packageFor$sys")
+}
+
+def setPackageProperties(String sys) {
```

Comment:
Aaahhhh, so after all my mentions of issues with the lists and indices, I think this method and all those lists can be setup with a simple map. Come and discuss.